generate_homepage_from_content.py

```diff
@@ -147,11 +150,15 @@ def html_element_from_content(content_dict):
     return elem.replace('%', '%%')
 
 
-def content_from_csv(filepath):
-    """Given the filepath of a csv containing content, returns a list of dicts,
-    each dict representing a row."""
-    return list(csv.DictReader(open(filepath)))
-
+def content_from_yaml(filepath):
+    """Given the filepath of a yaml file containing content, returns a dict of
+    content information."""
+    with open(filepath) as stream:
+      try:
+        contents = yaml.load(stream)
+      except yaml.YAMLError as ex:
+        print ex
```

Comment:
I don't see much point to catching this error if we don't add any additional info to it OR break the func. I would think we would WANT to `raise` the exception rather just printing the exception and potentially returning nonsense--which will either lead to more, weirder errors down the line, or us generating a blank content page and possibly overwriting a file.

I'd recommend just not catching the error--if something goes wrong, it'll raise an exception and crash the program--or if you are gonna catch it, just do so to add additional info (human readable string like "Something went wrong trying to read content from the yaml file!") and `raise` the exception anyway.